### DISCUSSION TITLE
Strict order in sales order / demmand priorities.

### DISCUSSION THREAD

**Discussion by salvastp** [frePPLe/frepple#609] [General]
Hello everyone,

I will try to explain my issue with an example:

I have 2 orders, A with priority 1 and B with priority 2,with the same due date and the same operations on the same resource. Order A still has a dependent operation on other resource, while B has no pending operations and can start immediately.
In my tests, the system plans order B first since it can be manufactured immediately and then order A when the pending operation is completed. 

The question is, can the system be forced to follow the strict order and manufacture order A first, making order B wait as long as necessary?

Thank you,

**Comment by jdetaeye** [comment]
> In my tests, the system plans order B first since it can be manufactured immediately and then order A when the pending operation is completed.

Which makes perfect sense to me :-)

> The question is, can the system be forced to follow the strict order and manufacture order A first, making order B wait as long as necessary?

If A and B are independent sales orders, why would you want this?  

If A and B are NOT independent, you will need to explicitly model their relationship. That relationship is normally modelled through an operation and its consumed materials or dependencies.

**Reply by salvastp** [reply]
Yes, A and B are independent. We want to start with A because the machine's current configuration is set for type A items, and it takes 10 hours to change the setup for type B. I'm not sure if we can define the current state of the machine's configuration.

**Reply by jdetaeye** [reply]
The precise sequencing of the *manufacturing orders* is a different topic than the sequencing/prioritization of the *sales orders*.   There are many reasons why your manufacturing orders are not sequenced in exactly the same ordering as your sales orders:   material constraints, different lead times, different resources in the supply path, etc... all can have an impact on this, especially in complex supply paths with many levels.

In your example, you need to think of generating "approved" or "confirmed" manufacturing orders on the critical resource with the long changeover time.   In other words: the algorithm (or the planner) put the manufacturing orders in a proper sequence, and frepple propagates the impact of that approved&confirmed plan to the sales orders.

> I'm not sure if we can define the current state of the machine's configuration.

https://demo.frepple.com/resource-setup-matrices/planningboard/ has an example with a detailed modeling of the changeover matrix on the resources, and the current state of the machine (in the "setup" field of the resource).
Caveat:   setup matrices are only available in the Enterprise & Cloud Editions, and not in Community Edition.
